File: 0055-jump-game/0055-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:

        currJump = 0
        for num in nums:
            if currJump < 0:
                return False
            currJump = max(currJump, num)
            currJump -= 1
   
        return True
        

File: 0055-jump-game/0055-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {

        int remainder = 0;

        for(int i=0; i<nums.length; i++){
            if(remainder < 0){
                return false;
            }

            if(remainder < nums[i]){
                remainder = nums[i];
            }
            remainder--;
        }

        return true;        
    }
}